VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/dao/ProviderDao.java

```diff
@@ -38,4 +49,69 @@ public long persistProvider(Provider provider) throws SQLException {
             throw new SQLException("Encountered problem creating a new provider ", e);
         }
     }
+
+    /**
+     *
+     * @param id of provider we're looking for
+     * @return list of providers with only 1 entry, when providerID is known
+     * @throws SQLException on SQL errors like table not existing
+     */
+    public List<Provider> getProvider(long id) throws SQLException {
+        try (PreparedStatement statement = connection.prepareStatement(FIND_BY_PROVIDER_ID_SQL)) {
+            statement.setLong(1, id);
+
+            ResultSet res = statement.executeQuery();
+            List<Provider> items = new ArrayList<>();
+
+            if(res.next() ){
+                extractProvider(res, items);
+            }
+            return items;
+        } catch (SQLException e) {
+            throw new SQLException("Encountered problem fetching a provider ", e);
+        }
+    }
+
+    /**
+     *
+     * @return ListOfAllProviders
+     * @throws SQLException When query is malformed, or other SQL related issues.
+     */
+    public List<Provider> getProviders() throws SQLException{
+        try (PreparedStatement statement = connection.prepareStatement(FIND_ALL_PROVIDERS_SQL)) {
+
+            ResultSet res = statement.executeQuery();
+            List<Provider> items = new ArrayList<>();
+
+            while(res.next() ){
+                extractProvider(res, items);
+            }
+            return items;
+        } catch (SQLException e) {
+            throw new SQLException("Encountered problem fetching providers ", e);
```

Comment:
In case of runtime exceptions we only work with the exceptions thrown by the system. Creating a new exception instance is not needed actually we already have the exception object. you can just print the stack trace using e.getMessage() along with a custom message like you just provided or don't handle it here if you plan to pass the exception to the upper classes in calling hierarchy.